cpp/src/gandiva/projector.cc

```diff
@@ -75,6 +92,17 @@ Status Projector::Make(SchemaPtr schema, const ExpressionVector& exprs,
   // Verify if previous projector obj code was cached
   if (prev_cached_obj != nullptr) {
     is_cached = true;
+  } else if (sec_cache != nullptr) {
+    std::shared_ptr<arrow::Buffer> arrow_buffer = sec_cache->Get(cache_key.Hash());
```

Comment:
Using just hash as key might lead to collisions..also you need to take into account the cpu features since the generated code will be different in different cpus.

Reply:
Added `Engine.Hash()` which includes the cpu attributes and name to the key